wagtailmedia/models.py

```diff
@@ -130,6 +150,31 @@ def get_media_model():
     return media_model
 
 
+# Receive the post_save signal and sniff mediainfo data if possible.
+@receiver(post_save, sender=Media)
+def media_sniff(sender, instance, created, update_fields, **kwargs):
+    if hasattr(settings, 'WAGTAILMEDIA_FFPROBE_CMD'):
+        if created or (update_fields and 'file' in update_fields):
+            data = sniff_media_data(instance.file.path)
+            if data:
+                duration = int(float(data['format']['duration']))
+                Media.objects.filter(pk=instance.pk).update(duration=duration, mediainfo=data)
```

Comment:
I agree with your comment that just dumping this `data` dictionary into a text field feels wasteful.

Reply:
I felt compelled to store the results of the media info extraction alongside the Media object. Maybe it should be called "raw_mediainfo" or something. I'm not sure how wasteful it is. Probably less wasteful than re-extracting via ffmpeg each time.

I don't love that it's a Textfield.... I'd prefer a JSONField but wanted to maximize compatibility.